Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeDelegate.cpp

```diff
@@ -226,13 +227,23 @@ namespace AzToolsFramework
             m_entityComponentIdPair = entityComponentIdPair;
             m_componentType = componentType;
 
+            AzFramework::ComponentModeDelegateNotificationBus::Broadcast(
+                &AzFramework::ComponentModeDelegateNotificationBus::Events::OnComponentModeDelegateConnect, m_entityComponentIdPair);
+
             EntitySelectionEvents::Bus::Handler::BusConnect(entityComponentIdPair.GetEntityId());
             EditorEntityVisibilityNotificationBus::Handler::BusConnect(entityComponentIdPair.GetEntityId());
             EditorEntityLockComponentNotificationBus::Handler::BusConnect(entityComponentIdPair.GetEntityId());
         }
 
         void ComponentModeDelegate::Disconnect()
         {
+            AzFramework::ComponentModeDelegateNotificationBus::Broadcast(
+                &AzFramework::ComponentModeDelegateNotificationBus::Events::OnComponentModeDelegateDisconnect, m_entityComponentIdPair);
```

Comment:
update: As we send this notification before the other bus connect calls, it would be good to call this after the bus disconnects in the Disconnect call (so Disconnect would look like this...)

```
EditorEntityLockComponentNotificationBus::Handler::BusDisconnect();
EditorEntityVisibilityNotificationBus::Handler::BusDisconnect();
EntitySelectionEvents::Bus::Handler::BusDisconnect();

AzFramework::ComponentModeDelegateNotificationBus::Broadcast(
     &AzFramework::ComponentModeDelegateNotificationBus::Events::OnComponentModeDelegateDisconnect, m_entityComponentIdPair);

m_componentType = AZ::Uuid::CreateNull();
m_entityComponentIdPair = AZ::EntityComponentIdPair(AZ::EntityId(), AZ::InvalidComponentId);
m_handler = nullptr;
```

Reply:
I'd still really prefer the disconnects to happen first (as in my snippet above) but this is okay for now

Reply:
Oh sorry I hadn't refreshed, looks good now 👍